vulnerabilities/importers/apache_httpd.py

```diff
@@ -81,56 +87,82 @@ def to_advisory(self, data):
                 for version_data in products["version"]["version_data"]:
                     versions_data.append(version_data)
 
-        fixed_version_ranges, affected_version_ranges = self.to_version_ranges(versions_data)
+        fixed_versions = []
+        for timeline_object in data["timeline"]:
+            timeline_value = timeline_object["value"]
+            if "release" in timeline_value:
+                split_timeline_value = timeline_value.split(" ")
```

Comment:
Quite often using partition() and rpartition() is more robust and readable than a split() where you never know how many splits and items you will get

Reply:
Working on @pombredanne 's last unaddressedcomment now.  I'm not familiar with `partition()` and `rpartition()` so researching atm.

Reply:
At first blush I don't think we want either `partition` or `rpartition` since our data is varied and these functions limit us to results split into 3 pieces, even if the split/partition target occurs multiple times in the target string.  Not even sure when this would be useful, but looking more closely into the `timeline` data we have with `apache_httpd` to check the results of these 2 functions vs. the straightforward `split()` function I've been using.

I've been splitting on `" "` and testing the contents of results[0] and results[-1].  Both `partition` and `rpartition` seem to alter the [0] and [-1] segments and render them unusable.  But . . . checking.

Reply:
@pombredanne Based on my testing I conclude that `partition` alters the results such that we can't use it to ID and report the fixed versions.  It looks like we can use `rpartition`, though the results differ from using `split` and I don't see the advantage.  I think we should stick with `split`.  What is your view?

Example:

Input:

```
timeline_values = ["1.3.14 released", "released in 2.4.54", "1.3-never released"]

for t_value in timeline_values:
    print('t_value.split(" ") = {}'.format(t_value.split(" ")))
    print('t_value.partition(" ") = {}'.format(t_value.partition(" ")))
    print('t_value.rpartition(" ") = {}'.format(t_value.rpartition(" ")))
    print("")
```

Output:

```
t_value.split(" ") = ['1.3.14', 'released']
t_value.partition(" ") = ('1.3.14', ' ', 'released')
t_value.rpartition(" ") = ('1.3.14', ' ', 'released')

t_value.split(" ") = ['released', 'in', '2.4.54']
t_value.partition(" ") = ('released', ' ', 'in 2.4.54')
t_value.rpartition(" ") = ('released in', ' ', '2.4.54')

t_value.split(" ") = ['1.3-never', 'released']
t_value.partition(" ") = ('1.3-never', ' ', 'released')
t_value.rpartition(" ") = ('1.3-never', ' ', 'released')
```

Reply:
@pombredanne @TG1999 Regardless of whether we stick with `split` or replace with `rpartition` (though no idea why we'd do that ;-), we don't have a test for this and I'm inclined to add one -- or is that overkill?

Reply:
I think I can tie up that loose end by adding one more JSON input, one with the `released in 2.4.54` syntax.  I'm going to go ahead and do that, stick with `split()` for now, and commit/push.  I can modify further if/when you gentlemen suggest.